syntaxes/alex.YAML-tmLanguage

```diff
@@ -0,0 +1,75 @@
+fileTypes:
+      - x
+name: Haskell Alex
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  comments:
+    patterns:
+      - name: comment.line
+        match: \-\-.*$
+      - name: comment.block
+        begin: "{-"
+        end: "-}"
```

Comment:
This doesn't handle nested comments. I recommend just copying the code for Haskell comments, e.g. just include the following:


```yaml
  block_comment:
    applyEndPatternLast: 1
    begin: "{-"
    captures:
      '0': {name: punctuation.definition.comment.alex}
    end: "-}"
    name: comment.block.alex
    patterns:
      - include: "#block_comment"
  comments:
    patterns:
      - include: "#block_comment"
      - name: comment.line.alex
        match: '\-\-.*$'
```

Reply:
Sorry, I think you'll want the "line comment" appear before the block comment, to properly handle code like

```haskell
-- comment {-
not a comment
-- comment -}